src/models/BanList.ts

```diff
@@ -107,8 +214,13 @@ export default class BanList {
             if (!entity || !recommendation) {
                 continue;
             }
-
-            this.rules.push(new ListRule(entity, recommendation, reason, kind));
+            const rule = new ListRule(entity, recommendation, reason, kind);
+            event.unsigned.rule = rule;
+            if (changeType) {
+                changes.push({rule, changeType, event, sender: event.sender, ... previousState ? {previousState} : {} });
+            }
+            this.rules.push(rule);
```

Comment:
We evict all of the rules from `this.rules` at the start of the update method and we're doing this to add them all back. See https://github.com/matrix-org/mjolnir/pull/143#discussion_r754530286 it might be silly that we still do this now.